### PR TITLE
Multiple domains can now make use of View hleper class

### LINKED ISSUE #35: A domain that is not the default domain cannot make use of the View helper class
## User Story
When building an add-on with multiple domains, any domain that is not the "default" domain, cannot make use of the `View` helper class.

## Details

Currently, only the "default" domain can make use of the View helper class. Trying to load the template file from another domain will throw an error or load the wrong file because the path to the domain resources contains "default" domain name.

## Expected Behavior

Every add-on domain should be able to make use of the `View` helper class.

## Steps to Reproduce

1. Generate a new add-on
2. Add a new domain directory named `TestDomain` in `src/` directory
3. Create a new template file named `testdomain.php` in `TestDomain/resources/views/` directory
4. Try to load `testdomain.php` template using `View` helper class

## Acceptance Criteria

-   [x] Every add-on domain is able to make use of the `View` helper class.

### PR DESCRIPTION
**Pull request by alaca**
Resolves #35

## Description

When building an add-on with multiple domains, any domain that is not the "default" domain, cannot make use of the `View` helper class. This PR resolves this issue by enabling developers to set domain paths within the View class template name argument.

The domain directory path can be set by using `.` in the template name.
Example: `View::render( 'DomainName.templateName' );`
String before the `.` character is domain directory, and everything after is the template file path.
This will try to load src/`DomainName`/resources/view/`templateName`.php file

## Affects
View helper class

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@since` tags included in DocBlocks
-   [x] Changes logged to the `Unreleased` section of `CHANGELOG.md`
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

## Testing Instructions

1. Generate a new add-on
2. Add a new domain directory named `TestDomain` in `src/` directory
3. Create a new template file named `testdomain.php` in `TestDomain/resources/views/` directory
4. Try to load `testdomain.php` template using `View::render('TestDomain.testdomain')`

### CODE REVIEW

**Comment by JasonTheAdams** [comment]
@kjohnson Can you please review this — no rush — as I'm curious what your thoughts are.

File: src/Addon/View.php [hunk 1]
```diff
@@ -49,4 +60,24 @@ public static function load( $view, $templateParams = [], $echo = false ) {
 	public static function render( $view, $vars = [] ) {
 		static::load( $view, $vars, true );
```
Comment (kjohnson) [inline src/Addon/View.php:61]: I would recommend removing the `$echo` flag parameter from the `load` method signature in favour of applying the `echo` construct directly in the `render` method definition. This reduces complexity and more clearly defines the responsibility of each method.

**Review by kjohnson — COMMENTED** [review]
This is a good implementation of a global view loader, similar to how Laravel architects view templates. Given the architecture of Laravel, with `resources` at the top level _next to_ the main `app` directory, this global view loader makes sense.

However, given that we are moving to a Domain Driven architecture, I think we need to be careful about how shared resources are used, with a global view loader blurring the lines of the Domains.

That said, I'm undecided on what a better solution would look like and my attempts have resulted in a more complicated implementation of what is actually just a global view loader.

I've considered using a View Factory to build a View helper for a given Domain, but that is largely just a (more "pure") syntax difference that the (more "practical") dot notation.

Additionally, I've considered making the View helper an abstract class with a concrete definition in each Domain (as needed), but this adds the complexity of inheritance where dot notation is simple.

My preference would be for a View Factory which binds the resulting View helper to the service container, but the container is shared by the Domains, so I don't think this works in practice.

In conclusion, the dot notation is intuitive and simple, which is in contrast to the above mental exercise.

**Comment by alaca** [comment]
Another approach would be to instantiate the View class where is needed and pass the path argument to the class constructor or create a factory method. Example:
`View::fromPath( 'path/to' )->renderView( $args )`

I like the dot notation because it's simple and it forces me to have the same directory structure for resources across all domains.

**Comment by kjohnson** [comment]
@alaca I agree that your dot notation approach is simpler.

**Comment by JasonTheAdams** [comment]
@alaca Is this ready to merge?

**Comment by alaca** [comment]
@JasonTheAdams yes, this is ready to be merged.
